examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -19,24 +70,55 @@ angular.module("cloudberry.sidebar", ["cloudberry.common"])
       $scope.isHashTagOutdated = false;
     }
 
-    function sendSampleTweetsQuery() {
-      var sampleTweetsRequest = queryUtil.getSampleTweetsRequest(cloudberry.parameters);
+    function sendSampleTweetsQuery(offset=0) {
+      var parameters = cloudberry.parameters;
+      var sampleTweetsRequest = {
+          dataset: parameters.dataset,
+          filter: [{
+            "field": "text",
+            "relation": "contains",
+            "values": [$scope.keyword]
+          },
+          {
+            "field": "create_at",
+            "relation": "inRange",
+            "values": [ timeLowerBound, timeUpperBound]
+          }
+          ],
+          select: {
+            order: ["-create_at"],
+            limit: 10,
+            offset: 0,
+            field: ["create_at", "id", "user.id"]
+          }
+        };
       cloudberryClient.send(sampleTweetsRequest, function(id, resultSet) {
-        cloudberry.commonTweetResult = resultSet[0];
+        $scope.sampleTweets = $scope.sampleTweets.concat(resultSet[0]);//oldest tweet will be at front
       }, "sampleTweetsRequest");
       $scope.isSampleTweetsOutdated = false;
     }
 
-    function handleSidebarQuery() {
-
+    function handleSidebarQuery(){
       if ($scope.isHashTagOpen && $scope.isHashTagOutdated) {
         sendHashTagQuery();
       }
-
+    }
+  
+    $scope.$watch(function(){
```

Comment:
We don't use this `watch` way to get notified messages if the any query condition is changed any more. Let's use `moduleManager` to subscribe to the `event` of `CHANGE_SEARCH_KEYWORD` if you want to get the notification. But for this case, this part logic actually has already been implemented in this `sidebar` controller. The place that we need to add this `sendQueryLoop` should be in `sendSampleTweetsQuery` function, because that function will be called if any query condition (besides `keyword change`, also `time-range change`, `zoom` or `drag`) changes happen. Let's talk about it F2F.